### PR TITLE
Add Django 1.11 compatibility.

### PR DESCRIPTION
**Pull request by doctoryes**
Just for testing...

### CODE REVIEW

**Comment by clintonb** [comment]
Where are we using this package? API endpoints that need Bearer authentication can use our custom backend. See http://edx-drf-extensions.readthedocs.io/en/latest/settings.html#bearerauthentication.

**Comment by doctoryes** [comment]
`find edx-platform -name \*.py -exec grep rest_framework_oauth {} \; -print` shows 10 different places. Are you implying that it would be as simple as swapping out the `OAuth2Authentication` classes for `BearerAuthentication` classes?

**Comment by clintonb** [comment]
It should be that easy. You'll also need to configure the appropriate Django settings for that class.

**Comment by doctoryes** [comment]
@clintonb I've tried creating a PR that removes this module as a dependency. The work-in-progress is here:
https://github.com/edx/edx-platform/pull/15603
However, it's incomplete and missing the Django settings configuration you mention.
Is the work to remove this dependency represented as a JIRA ticket? If not, I'd be happy to create a ticket. I'd like to move ahead with merging this PR and using this update in edx-platform for the Django 1.11 upgrade work. It'd be great to remove this dependency altogether, but it's out of scope of this work.

**Comment by clintonb** [comment]
There is no ticket. Please create one.

File: setup.py [hunk 1]
```diff
@@ -91,7 +91,6 @@ def get_package_data(package):
     packages=get_packages(package),
     package_data=get_package_data(package),
     cmdclass={'test': PyTest},
-    install_requires=install_requires,
```
Comment (bmedx) [inline setup.py:94]: It looks like there are valid requirements, is there a reason they're not shown here?

**Comment by clintonb** [comment]
Can we expand the scope of the Django 1.11 project to ditch this package, please? Given that it is pretty much dead (https://github.com/jpadilla/django-rest-framework-oauth/issues/11), I don't think we should invest in this package any longer.

**Review by clintonb — APPROVED** [review]
I disagree with continuing to use this package, but this PR is fine as-is.

**Review by rlucioni — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (doctoryes) [reply setup.py:94]: The `install_requires` were removed in @rlucioni 's PR to make the dependencies optional based on which auth method was being used:
https://github.com/jpadilla/django-rest-framework-oauth/pull/12

**Comment by doctoryes** [comment]
I've created this ticket to perform the work to remove this dependency from edx-platform:
https://openedx.atlassian.net/browse/PLAT-1660

File: setup.py (hunk 1 above)
Comment (rlucioni) [reply setup.py:94]: @doctoryes if you're removing this you should also remove the line above that assigns `install_requires`. Out of curiosity, why aren't those changes already on this branch?

Comment (doctoryes) [reply setup.py:94]: That was poor merging on my part between the changes made on the edx fork versus upstream. I'll make a PR to remove that `install_requires` - though it won't actually affect anything in this version.

**Pull request merged**